images/egress/dns-proxy/egress-dns-proxy.sh

```diff
@@ -0,0 +1,183 @@
+#!/bin/bash
+
+# OpenShift egress DNS proxy setup script
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+# Default DNS nameserver port
+NS_PORT=53
+CONF=/etc/haproxy/haproxy.cfg
+
+BLANK_LINE_OR_COMMENT_REGEX="([[:space:]]*$|#.*)"
+IPADDR_REGEX="[[:digit:]]+\.[[:digit:]]+\.[[:digit:]]+\.[[:digit:]]+"
+PORT_REGEX="[[:digit:]]+"
+DOMAINNAME_REGEX="[[:alnum:]][[:alnum:]-]+?\.[[:alnum:].-]+"
+
+function die() {
+    echo "$*" 1>&2
+    exit 1
+}
+
+function check_prereqs() {
+    if [[ -z "${EGRESS_SOURCE}" ]]; then
+        die "Must specify EGRESS_SOURCE"
+    fi
+    if ! [[ "${EGRESS_SOURCE}" =~ ^${IPADDR_REGEX} ]]; then
+        die "EGRESS_SOURCE must be IPv4 address"
+    fi
+
+    if [[ -z "${EGRESS_DESTINATION}" ]]; then
+        die "Must specify EGRESS_DESTINATION"
+    fi
+}
+
+function validate_port() {
+    local port=$1
+    if [[ "${port}" -lt "1" || "${port}" -gt "65535" ]]; then
+        die "Invalid port: ${port}, must be in the range 1 to 65535"
+    fi
+}
+
+function generate_haproxy_defaults() {
+    echo "
+global
+    log         127.0.0.1 local2
+
+    chroot      /var/lib/haproxy
+    pidfile     /var/run/haproxy.pid
+    maxconn     4000
+    user        haproxy
+    group       haproxy
+
+defaults
+    log                     global
+    mode                    tcp
+    option                  dontlognull
+    option                  tcplog
+    option                  redispatch
+    retries                 3
+    timeout http-request    100s
+    timeout queue           1m
+    timeout connect         10s
+    timeout client          1m
+    timeout server          1m
+    timeout http-keep-alive 100s
+    timeout check           10s
+"
+}
+
+function generate_dns_resolvers() {
+    echo "resolvers dns-resolver"
+    # Fetch nameservers from /etc/resolv.conf
+    declare -a nameservers=$(cat /etc/resolv.conf |grep nameserver|awk -F" " '{print $2}')
```

Comment:
origin style is to put declaration and initialization on separate lines. (IIRC because if you write it the way you did here, and there's an error, the error gets ignored)

also, you don't need cat and grep: `$(awk '/nameserver/ {print $2}' /etc/resolv.conf)`

also, actually you need a `^` before `nameserver` so it doesn't get tripped up by the comments NetworkManager sometimes adds:

    # NOTE: the libc resolver may not support more than 3 nameservers.
    # The nameservers listed below may not be recognized.